Source/WebCore/platform/network/cocoa/ResourceRequestCocoa.mm

```diff
@@ -64,7 +64,8 @@
 ResourceRequest::ResourceRequest(ResourceRequestPlatformData&& platformData, const String& cachePartition, bool hiddenFromInspector)
 {
     if (platformData.m_urlRequest) {
-        setRequester(*platformData.m_requester);
+        if (platformData.m_requester)
+            setRequester(*platformData.m_requester);
         m_nsRequest = platformData.m_urlRequest;
         setIsAppInitiated(*platformData.m_isAppInitiated);
```

Comment:
The `m_isAppInitiated` field is also `std::optional<>`, so this could also crash due to `platformData.m_isAppInitiated` not being set:

```
    std::optional<bool> m_isAppInitiated;
```

One way to fix this one is to use `std::optional<>::value_or()` (assuming `false` should be the default value if the `std::optional<>` is not set):

```suggestion
        setIsAppInitiated(platformData.m_isAppInitiated.value_or(false));
```

You don't have to fix this with this PR, but please fix this in a follow-up as we can't assume a `std::optional<>` is always set like this.